mesonbuild/mesonlib.py

```diff
@@ -972,3 +980,23 @@ def update(self, iterable):
 
     def difference(self, set_):
         return type(self)(e for e in self if e not in set_)
+
+class BuildDirLock:
+
+    def __init__(self, builddir):
+        self.lockfilename = os.path.join(builddir, 'meson-private/meson.lock')
+
+    def __enter__(self):
+        if not have_fcntl:
+            return
+        self.lockfile = open(self.lockfilename, 'w')
```

Comment:
'x'?

Reply:
That always creates a new file (according to docs at least). We want to use the existing file when it is there (which is most of the time).

Reply:
But checking for file existence instead of a lock would work on Windows, no?

Reply:
I _think_ (but am not sure) simply opening the file works as a lock on Windows because only one thing can open a file at a time so we could do that on Windows. If someone knows it that works for sure, let me know.